android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -648,6 +773,101 @@ void createTouchScalar(ImageResolution resolution, DisplayMetrics displayMetrics
         return motionEventList;
     }
 
+    public VideoStreamingParameters getLastCachedStreamingParameters() {
+        return parameters;
+    }
+
+    private List<VideoStreamingCapability> getSupportedCapabilities(
+            List<VideoStreamingRange> ranges,
+            List<VideoStreamingCapability> originalAdditionalCapabilities
+    ){
+
+        List<VideoStreamingCapability> validCapabilities = new ArrayList<>();
+
+        VideoStreamingCapability preferredCapability = new VideoStreamingCapability();
+        preferredCapability.setDiagonalScreenSize(parameters.getPreferredDiagonal());
+        preferredCapability.setPreferredResolution(new ImageResolution(
+                parameters.getResolution().getResolutionWidth(),
+                parameters.getResolution().getResolutionHeight())
+        );
+
+        // get the first one - the Desired resolution to guarantee streaming will start
+        validCapabilities.add(preferredCapability);
+        for (VideoStreamingRange range: ranges) {
+
+            Integer constraintHeightMax = range.getMaxResolution().getResolutionHeight();
+            Integer constraintHeightMin = range.getMinResolution().getResolutionHeight();
+
+            for (VideoStreamingCapability capability : originalAdditionalCapabilities) {
+                double diagonal;
+                if (capability.getPreferredResolution() == null
+                        || capability.getPreferredResolution().getResolutionHeight() == null
+                        || capability.getPreferredResolution().getResolutionWidth() == null) {
+                    continue;
+                }
+                if (capability.getDiagonalScreenSize() == null) {
+                    diagonal = parameters.getPreferredDiagonal();
+                } else {
+                    diagonal = capability.getDiagonalScreenSize();
+                }
+
+                if (range.getMinScreenDiagonal() > diagonal) {
+                    continue;
+                }
+
+                if (!isAspectRatioInRange(range.getAspectRatio(), capability.getPreferredResolution())) {
+                    if (constraintHeightMax == null && constraintHeightMin == null) {
+                        continue;
+                    }
+                }
+
+                if (!isImageResolutionInRange(range.getMinResolution(), range.getMaxResolution(), capability.getPreferredResolution())) {
+                    continue;
+                }
+
+                validCapabilities.add(capability);
+            }
+        }
+
+        return validCapabilities;
+    }
+
+    public Boolean isImageResolutionInRange(Resolution minResolution, Resolution maxResolution, ImageResolution currentResolution) {
```

Comment:
@kostyaBoss Per the proposal, `isImageResolutionInRange` should be defined as `public Boolean isImageResolutionInRange(VideoStreamingRange range, ImageResolution currentResolution)`.

Your definition is different because you take in two `Resolution` objects instead of just the `VideoStreamingRange` object. Do you need to change yours to match up with the proposal?

Reply:
@santanamk in the proposal it was just a section where a preview example was provided. To keep libraries mirroring each other we've agreed to make startRemoteDisplayStream with two arguments